File: PortfolioTracker.API/Entities/MarketPrice.cs
using System.ComponentModel.DataAnnotations;

namespace PortfolioTracker.API.Entities;

public class MarketPrice
{
    [Key]
    [MaxLength(120)]
    public string Symbol { get; set; } = string.Empty;

    [MaxLength(140)]
    public string ProviderSymbol { get; set; } = string.Empty;

    [MaxLength(240)]
    public string? CompanyName { get; set; }

    public decimal? CurrentPrice { get; set; }

    public decimal? DayHigh { get; set; }

    public decimal? DayLow { get; set; }

    public decimal? MarketCap { get; set; }

    public DateTime? FetchedAt { get; set; }

    public bool IsAvailable { get; set; }

    public bool IsManual { get; set; }

    public DateTime? ManualUpdatedAt { get; set; }

    [MaxLength(500)]
    public string? Error { get; set; }
}

File: PortfolioTracker.Frontend/src/types/marketPrice.ts
export interface MarketPriceQuote {
  symbol: string;
  companyName?: string | null;
  currentPrice?: number | null;
  dayHigh?: number | null;
  dayLow?: number | null;
  marketCap?: number | null;
  fetchedAt?: string | null;
  delayMinutes?: number | null;
  isAvailable: boolean;
  isManual: boolean;
  isRefreshing: boolean;
  manualUpdatedAt?: string | null;
  error?: string | null;
}
